src/sidebars/favourites/favourites.js

```diff
@@ -2,117 +2,126 @@
-                }
+                    // Add new stocks from favourites
+                    self.favourites.map(function(favourite) {
+                        if (self.stocks.map(function(stock) { return stock.code; }).indexOf(favourite) === -1) {
+                            // This is a new stock
+                            quandlService.getData(favourite, function(stock) {
+                                var data = stock.data[0],
+                                    price,
+                                    delta,
+                                    percentage;
+
+                                if (data) {
+                                    price = data.close;
+                                    delta = data.close - data.open;
+                                    percentage = delta / data.open * 100;
+
+                                    self.stocks.push({
+                                        name: stock.name,
+                                        code: stock.code,
+                                        price: price,
+                                        delta: delta,
+                                        percentage: Math.abs(percentage),
+                                        notification: false,
+                                        favourite: true
+                                    });
+                                }
+                            });
+                        }
+                    })
+                };
 
-                // Add new stocks from favourites
-                self.favourites.map(function(favourite) {
-                    if (self.stocks.map(function(stock) { return stock.code; }).indexOf(favourite) === -1) {
-                        // This is a new stock
-                        quandlService.getData(favourite, function(stock) {
-                            var data = stock.data[0],
-                                price,
-                                delta,
-                                percentage;
-
-                            if (data) {
-                                price = data.close;
-                                delta = data.close - data.open;
-                                percentage = delta / data.open * 100;
-
-                                self.stocks.push({
-                                    name: stock.name,
-                                    code: stock.code,
-                                    price: price,
-                                    delta: delta,
-                                    percentage: Math.abs(percentage),
-                                    notification: false
-                                });
-                            }
-                        });
-                    }
-                })
-            };
+                self.update();
 
-            self.updateFavourites();
-        }])
+                $scope.$on('favouriteChanged', function(event, data) {
```

Comment:
It actually is more cross cutting than i thought. I think event is the right way to go.